src/module.c

```diff
@@ -5954,8 +5954,10 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
             sds object = (acl_retval == ACL_DENIED_CMD) ? sdsdup(c->cmd->fullname) : sdsdup(c->argv[acl_errpos]->ptr);
             addACLLogEntry(ctx->client, acl_retval, ACL_LOG_CTX_MODULE, -1, ctx->client->user->name, object);
             if (error_as_call_replies) {
-                sds msg = sdscatfmt(sdsempty(), "acl verification failed, %s.", getAclErrorMessage(acl_retval));
-                reply = callReplyCreateError(msg, ctx);
+                sds acl_msg = getAclErrorMessage(acl_retval, ctx->client->user, c->cmd, argv, acl_errpos);
+                sds msg = sdscatfmt(sdsempty(), "-NOPERM %s\r\n", acl_msg);
+                sdsfree(acl_msg);
```

Comment:
It's important in `RM_Call()` to use `callReplyCreateError()` for errors, to have this information for `RM_ReplyWithCallReply()` calls.

Reply:
callReplyCreateError doesn't work for -NOPERM response, as it prepends an -ERR string.  perhaps we need callReplyCreateError that takes a type (i.e. a string), which in general will be "ERR", but can select as NOPERM.

thoughts?

Reply:
no, it will add `-ERR` only in case the reply doesn't start with `-`

Reply:
ok.  done